core/src/org/pentaho/di/core/database/Database.java

```diff
@@ -2173,6 +2157,73 @@ public RowMetaInterface getQueryFields( String sql, boolean param, RowMetaInterf
     return fields;
   }
 
+  private RowMetaInterface getQueryFieldsFromPreparedStatement( String sql, boolean param, RowMetaInterface inform,
+      Object[] data ) throws Exception {
+    PreparedStatement preparedStatement = null;
+    try {
+      preparedStatement =
+          connection.prepareStatement( databaseMeta.stripCR( sql ), ResultSet.TYPE_FORWARD_ONLY,
+              ResultSet.CONCUR_READ_ONLY );
+      preparedStatement.setMaxRows( 1 );
+      ResultSetMetaData rsmd = preparedStatement.getMetaData();
+      return getRowInfo( rsmd, false, false );
+    } catch ( Exception e ) {
+      throw new Exception( e );
+    } finally {
+      if ( preparedStatement != null ) {
+        try {
+          preparedStatement.close();
+        } catch ( SQLException e ) {
+          throw new KettleDatabaseException( "Unable to close prepared statement after determining SQL layout", e );
+        }
+      }
+    }
+  }
+
+  private RowMetaInterface getQueryFieldsFromDatabaseMetaData( String sql, boolean param, RowMetaInterface inform,
```

Comment:
@e-cuellar just being extra picky, but how many of these parameters are actually used in this method?  Can we chop it down to only the ones used?